### ISSUE TITLE
signature: tracking issue for derive(Signer) and derive(Verifier) stabilization

### ISSUE THREAD

**Issue by tarcieri** [RustCrypto/traits#93]
The `signature` crate contains (as re-exports from the `signature_derive` crate) the following procedural macros which are presently gated under the `derive-preview` feature:

- [`derive(Signer)`](https://docs.rs/signature/1.0.0-pre.5/signature/derive.Signer.html) - support for deriving [`Signature`](https://docs.rs/signature/1.0.0-pre.5/signature/trait.Signature.html) on types which implement [`DigestSignature`](https://docs.rs/signature/1.0.0-pre.5/signature/trait.DigestSignature.html) for signature types which are also marked as [`DigestSignature`](https://docs.rs/signature/1.0.0-pre.5/signature/trait.DigestSignature.html)
- [`derive(Verifier)`](https://docs.rs/signature/1.0.0-pre.5/signature/derive.Verifier.html) - support for deriving [`Verifier`](https://docs.rs/signature/1.0.0-pre.5/signature/trait.Verifier.html) on types which implement [`DigestVerifier`](https://docs.rs/signature/1.0.0-pre.5/signature/trait.DigestVerifier.html) for signature types which are also marked as [`DigestSignature`](https://docs.rs/signature/1.0.0-pre.5/signature/trait.DigestSignature.html)

The main blocker for stabilization of this feature is #92: stabilizing the other digest-related functionality the proc macro builds on, and ongoing design questions.

**Comment by tarcieri** [comment]
We landed on a proc macro for this purpose after a lot of experimentation with alternative approaches, including blanket impls w\ marker traits. Some past discussion can be found here:

- [RustCrypto/signatures#9: Add UseDigestTo* marker trait and blanket Sign/Verify impls](https://github.com/RustCrypto/signatures/pull/9)
- [RustCrypto/signatures#10: Add "raw digest" signature marker trait and sign/verify traits](https://github.com/RustCrypto/signatures/pull/10)
- [RustCrypto/signatures#12: Digestable signatures](https://github.com/RustCrypto/signatures/pull/12)
- [RustCrypto/signatures#18: signature_derive: Custom derive support for Signer/Verifier](https://github.com/RustCrypto/signatures/pull/18)
- [RustCrypto/signatures#27: Replace DigestSignature with derive attr](https://github.com/RustCrypto/signatures/pull/27)
- [RustCrypto/signatures#33: Revert removal of DigestSignature](https://github.com/RustCrypto/signatures/pull/33)

The tl;dr: of all of these PRs is we've explored a lot of the potential design space and alternatives to a proc macro, netting out where we are because none of them were flexible enough to cover all of the potential use cases.

**Comment by tarcieri** [comment]
Stabilized in #1141
